plugins/spark/v3.5/integration/src/intTest/java/org/apache/polaris/spark/quarkus/it/SparkIT.java

```diff
@@ -30,175 +30,190 @@
+      assertThat(tables.size()).isEqualTo(2);
+      assertThat(tables)
+          .contains(
+              new Object[] {namespace, icebergTable, false},
+              new Object[] {namespace, deltaTable, false});
+
+      // verify the table and view content
+      List<Object[]> results = sql("SELECT * FROM %s ORDER BY col1", icebergTable);
+      assertThat(results.size()).isEqualTo(2);
+      assertThat(results.get(0)).isEqualTo(new Object[] {1, "a"});
+      assertThat(results.get(1)).isEqualTo(new Object[] {2, "b"});
+
+      // verify the table and view content
+      results = sql("SELECT * FROM %s ORDER BY col1", viewName);
+      assertThat(results.size()).isEqualTo(2);
+      assertThat(results.get(0)).isEqualTo(new Object[] {3, "a"});
+      assertThat(results.get(1)).isEqualTo(new Object[] {4, "b"});
+
+      List<Object[]> views = sql("SHOW VIEWS");
+      assertThat(views.size()).isEqualTo(1);
+      assertThat(views).contains(new Object[] {namespace, viewName, false});
+
+      // drop views and tables
+      sql("DROP TABLE %s", icebergTable);
+      sql("DROP TABLE %s", deltaTable);
+      sql("DROP VIEW %s", viewName);
+      sql("DROP NAMESPACE %s", namespace);
+
+      // clean up delta directory
+      File dirToDelete = new File(deltaDir);
+      FileUtils.deleteQuietly(dirToDelete);
+    }
   }
 
-  @Test
-  public void renameIcebergViewAndTable() {
-    String namespace = generateName("ns");
-    sql("CREATE NAMESPACE %s", namespace);
-    sql("USE %s", namespace);
-
-    // create one view and one table
-    String viewName = "originalView";
-    sql("CREATE VIEW %s AS SELECT 1 AS id", viewName);
-
-    String icebergTable = "iceberg_table";
-    sql("CREATE TABLE %s (col1 int, col2 string)", icebergTable);
-
-    // verify view and table is showing correctly
-    List<Object[]> views = sql("SHOW VIEWS");
-    assertThat(views.size()).isEqualTo(1);
-    assertThat(views).contains(new Object[] {namespace, viewName, false});
-
-    List<Object[]> tables = sql("SHOW TABLES");
-    assertThat(tables.size()).isEqualTo(1);
-    assertThat(tables).contains(new Object[] {namespace, icebergTable, false});
-
-    // rename the view
-    String renamedView = "renamedView";
-    sql("ALTER VIEW %s RENAME TO %s", viewName, renamedView);
-    views = sql("SHOW VIEWS");
-    assertThat(views.size()).isEqualTo(1);
-    assertThat(views).contains(new Object[] {namespace, renamedView, false});
-
-    // rename the table
-    String newIcebergTable = "iceberg_table_new";
-    sql("ALTER TABLE %s RENAME TO %s", icebergTable, newIcebergTable);
-    tables = sql("SHOW TABLES");
-    assertThat(tables.size()).isEqualTo(1);
-    assertThat(tables).contains(new Object[] {namespace, newIcebergTable, false});
-
-    // clean up the resources
-    sql("DROP VIEW %s", renamedView);
-    sql("DROP TABLE %s", newIcebergTable);
-    sql("DROP NAMESPACE %s", namespace);
+  @QuarkusIntegrationTest
+  static class S3ATest extends BaseTest {
```

Comment:
Ah, yeah, found it. Let's not duplicate all these tests, that's quite expensive. Can we just do one smoke test reading/writing a table?

Reply:
Sure that makes sense. Let me add one test that does the basic and cross check

Reply:
I have reverted this and added a new test `PolarisS3InterOperabilityTest.java`